Systems/c172p-engine.xml

```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0"?>
+
+<!--
+
+  c172p damage effects for JSBSim.
+
+    Copyright (c) 2015 Gilberto Agostinho
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+
+-->
+
+<system name="engine">
+
+    <channel name="oil">
+    
+        <fcs_function name="extra-weight-oil">
+            <function>
+                <product>
+                    <property>/engines/active-engine/oil-level</property>
+                    <value>1.7686</value>
+                </product>
+            </function>
+            <output>/fdm/jsbsim/inertia/pointmass-weight-lbs[16]</output>
+        </fcs_function>
+        
+    </channel>
+    
+    <channel name="carburetor">
+    
+        <fcs_function name="Carb heat on causes RPM drop on 160 HP engine">
+            <function>
+                <ifthen>
+                    <property>/controls/engines/current-engine/carb-heat</property>
```

Comment:
Okay, so I fixed it but there is a problem: as I mentioned before, the 180 HP engine's carb-heat is being controlled by `/controls/anti-ice/engine[0]/carb-heat` rather than `/controls/anti-ice/engine[1]/carb-heat`. So now this code is pretty but the carb heat does not drop RPM in the 180 HP engine. 

I also noticed that the flight recorder only saves the property `/controls/anti-ice/engine[0]/carb-heat` and not `/controls/anti-ice/engine[1]/carb-heat`. So what should I do, how can this be fixed?